doc/developer/man/man3/BMainWindow.3
.TH "BMainWindow" 3 "21 Apr 2009" "Version 0.1" "BrainCognize" \" -*- nroff -*-
.ad l
.nh
.SH NAME
BMainWindow \- This class contains the main window and the layout of the widgets within the window.  

.PP
.SH SYNOPSIS
.br
.PP
\fC#include <BMainWindow.h>\fP
.PP
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "void \fBsetupGUI\fP (QMainWindow *MainWindow)"
.br
.RI "\fISet up all widgets menus an buttons in the main window. \fP"
.in -1c
.SS "Private Attributes"

.in +1c
.ti -1c
.RI "QAction * \fBm_actionLoad\fP"
.br
.ti -1c
.RI "QAction * \fBm_actionSave\fP"
.br
.ti -1c
.RI "QAction * \fBm_actionPreferences\fP"
.br
.ti -1c
.RI "QAction * \fBm_actionExit\fP"
.br
.ti -1c
.RI "QAction * \fBm_actionAbout_BrainCognize\fP"
.br
.ti -1c
.RI "QWidget * \fBm_centralwidget\fP"
.br
.ti -1c
.RI "QMenuBar * \fBm_menubar\fP"
.br
.ti -1c
.RI "QMenu * \fBm_menuFile\fP"
.br
.ti -1c
.RI "QMenu * \fBm_menuHelp\fP"
.br
.ti -1c
.RI "QStatusBar * \fBm_statusBar\fP"
.br
.ti -1c
.RI "QToolBar * \fBm_toolBar\fP"
.br
.ti -1c
.RI "\fBBQtGLWidget\fP * \fBm_glWidget\fP"
.br
.ti -1c
.RI "\fBBQtPipelineBrowser\fP * \fBm_pipelineBrowser\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
This class contains the main window and the layout of the widgets within the window. 
.PP
Definition at line 29 of file BMainWindow.h.
.SH "Member Function Documentation"
.PP 
.SS "void BMainWindow::setupGUI (QMainWindow * MainWindow)"
.PP
Set up all widgets menus an buttons in the main window. 
.PP
Definition at line 14 of file BMainWindow.cpp.
.PP
References m_actionAbout_BrainCognize, m_actionExit, m_actionLoad, m_actionPreferences, m_actionSave, m_centralwidget, m_glWidget, m_menubar, m_menuFile, m_menuHelp, m_pipelineBrowser, m_statusBar, and m_toolBar.
.PP
Referenced by main().
.SH "Member Data Documentation"
.PP 
.SS "QAction* \fBBMainWindow::m_actionLoad\fP\fC [private]\fP"
.PP
Definition at line 39 of file BMainWindow.h.
.PP
Referenced by setupGUI().
.SS "QAction* \fBBMainWindow::m_actionSave\fP\fC [private]\fP"
.PP
Definition at line 40 of file BMainWindow.h.
.PP
Referenced by setupGUI().
.SS "QAction* \fBBMainWindow::m_actionPreferences\fP\fC [private]\fP"
.PP
Definition at line 41 of file BMainWindow.h.
.PP
Referenced by setupGUI().
.SS "QAction* \fBBMainWindow::m_actionExit\fP\fC [private]\fP"
.PP
Definition at line 42 of file BMainWindow.h.
.PP
Referenced by setupGUI().
.SS "QAction* \fBBMainWindow::m_actionAbout_BrainCognize\fP\fC [private]\fP"
.PP
Definition at line 43 of file BMainWindow.h.
.PP
Referenced by setupGUI().
.SS "QWidget* \fBBMainWindow::m_centralwidget\fP\fC [private]\fP"
.PP
Definition at line 44 of file BMainWindow.h.
.PP
Referenced by setupGUI().
.SS "QMenuBar* \fBBMainWindow::m_menubar\fP\fC [private]\fP"
.PP
Definition at line 45 of file BMainWindow.h.
.PP
Referenced by setupGUI().
.SS "QMenu* \fBBMainWindow::m_menuFile\fP\fC [private]\fP"
.PP
Definition at line 46 of file BMainWindow.h.
.PP
Referenced by setupGUI().
.SS "QMenu* \fBBMainWindow::m_menuHelp\fP\fC [private]\fP"
.PP
Definition at line 47 of file BMainWindow.h.
.PP
Referenced by setupGUI().
.SS "QStatusBar* \fBBMainWindow::m_statusBar\fP\fC [private]\fP"
.PP
Definition at line 48 of file BMainWindow.h.
.PP
Referenced by setupGUI().
.SS "QToolBar* \fBBMainWindow::m_toolBar\fP\fC [private]\fP"
.PP
Definition at line 49 of file BMainWindow.h.
.PP
Referenced by setupGUI().
.SS "\fBBQtGLWidget\fP* \fBBMainWindow::m_glWidget\fP\fC [private]\fP"
.PP
Definition at line 51 of file BMainWindow.h.
.PP
Referenced by setupGUI().
.SS "\fBBQtPipelineBrowser\fP* \fBBMainWindow::m_pipelineBrowser\fP\fC [private]\fP"
.PP
Definition at line 52 of file BMainWindow.h.
.PP
Referenced by setupGUI().

.SH "Author"
.PP 
Generated automatically by Doxygen for BrainCognize from the source code.


src/gui/qt4/BMainWindow.cpp
//---------------------------------------------------------------------------
//
// Project: BrainCognize
//
// Copyright 2009 SomeCopyrightowner
//
// This file is part of BrainCognize.
//
// BrainCognize is free software: you can redistribute it and/or modify
// it under the terms of the GNU Lesser General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// BrainCognize is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public License
// along with BrainCognize. If not, see <http://www.gnu.org/licenses/>.
//
//---------------------------------------------------------------------------

#include <QtGui/QDockWidget>

#include "BMainWindow.h"
#include "BQtGLWidget.h"
#include "BQtPipelineBrowser.h"

#include "../icons/BIcons.h"

void BMainWindow::addDockableGLWidget( QMainWindow *MainWindow )
{
    m_glWidget = new BQtGLWidget();
    QDockWidget *dockWidget = new QDockWidget( "Graphics Display" );
    dockWidget->setAllowedAreas( Qt::LeftDockWidgetArea
            | Qt::RightDockWidgetArea );
    dockWidget->setWidget( m_glWidget );
    MainWindow->addDockWidget( Qt::LeftDockWidgetArea, dockWidget );
}

void BMainWindow::setupGUI( QMainWindow *MainWindow )
{
    QIcon mainWindowIcon;
    QIcon quitIcon;
    QIcon saveIcon;
    QIcon loadIcon;
    QIcon aboutIcon;

    mainWindowIcon.addPixmap( QPixmap( logoIcon_xpm ) );
    quitIcon.addPixmap( QPixmap( quit_xpm ) );
    saveIcon.addPixmap( QPixmap( disc_xpm ) );
    loadIcon.addPixmap( QPixmap( fileopen_xpm ) );
    aboutIcon.addPixmap( QPixmap( logoIcon_xpm ) );

    if( MainWindow->objectName().isEmpty() )
        MainWindow->setObjectName( QString::fromUtf8( "MainWindow" ) );
    MainWindow->resize( 946, 632 );
    MainWindow->setWindowIcon( mainWindowIcon );

    m_actionLoad = new QAction( MainWindow );
    m_actionLoad->setObjectName( QString::fromUtf8( "actionLoad" ) );
    m_actionLoad->setIcon( loadIcon );
    m_actionLoad->setText( QApplication::translate( "MainWindow", "Load", 0,
            QApplication::UnicodeUTF8 ) );
    m_actionLoad->setShortcut( QApplication::translate( "MainWindow", "Ctrl+L",
            0, QApplication::UnicodeUTF8 ) );

    m_actionSave = new QAction( MainWindow );
    m_actionSave->setObjectName( QString::fromUtf8( "actionSave" ) );
    m_actionSave->setIcon( saveIcon );
    m_actionSave->setText( QApplication::translate( "MainWindow", "Save", 0,
            QApplication::UnicodeUTF8 ) );

    m_actionPreferences = new QAction( MainWindow );
    m_actionPreferences->setObjectName( QString::fromUtf8( "actionPreferences" ) );
    m_actionPreferences->setText( QApplication::translate( "MainWindow",
            "Preferences", 0, QApplication::UnicodeUTF8 ) );

    m_actionExit = new QAction( MainWindow );
    m_actionExit->setObjectName( QString::fromUtf8( "actionExit" ) );
    m_actionExit->setIcon( quitIcon );
    m_actionExit->setText( QApplication::translate( "MainWindow", "E&xit", 0,
            QApplication::UnicodeUTF8 ) );
    m_actionExit->setShortcut( QApplication::translate( "MainWindow", "Ctrl+Q",
            0, QApplication::UnicodeUTF8 ) );

    m_actionAbout_BrainCognize = new QAction( MainWindow );
    m_actionAbout_BrainCognize->setObjectName( QString::fromUtf8(
            "actionAbout_BrainCognize" ) );
    m_actionAbout_BrainCognize->setIcon( aboutIcon );
    m_actionAbout_BrainCognize->setText( QApplication::translate( "MainWindow",
            "About BrainCognize", 0, QApplication::UnicodeUTF8 ) );

    m_centralwidget = new QWidget( MainWindow );
    m_centralwidget->setObjectName( QString::fromUtf8( "centralwidget" ) );
    MainWindow->setCentralWidget( m_centralwidget );

    m_menubar = new QMenuBar( MainWindow );
    m_menubar->setObjectName( QString::fromUtf8( "menubar" ) );
    m_menubar->setGeometry( QRect( 0, 0, 946, 24 ) );

    m_menuFile = new QMenu( m_menubar );
    m_menuFile->setObjectName( QString::fromUtf8( "menuFile" ) );

    m_menuHelp = new QMenu( m_menubar );
    m_menuHelp->setObjectName( QString::fromUtf8( "menuHelp" ) );
    MainWindow->setMenuBar( m_menubar );

    m_toolBar = new QToolBar( MainWindow );
    m_toolBar->setObjectName( QString::fromUtf8( "toolBar" ) );
    m_toolBar->addAction( m_actionExit );
    m_toolBar->addAction( m_actionSave );
    m_toolBar->addAction( m_actionLoad );
    m_toolBar->addSeparator();
    m_toolBar->addAction( m_actionAbout_BrainCognize );
    MainWindow->addToolBar( Qt::TopToolBarArea, m_toolBar );

    m_statusBar = new QStatusBar( MainWindow );
    m_statusBar->setObjectName( QString::fromUtf8( "statusBar" ) );
    m_statusBar->showMessage( "No status message yet." );
    MainWindow->setStatusBar( m_statusBar );

    m_menubar->addAction( m_menuFile->menuAction() );
    m_menubar->addAction( m_menuHelp->menuAction() );

    m_menuFile->addAction( m_actionLoad );
    m_menuFile->addAction( m_actionSave );
    m_menuFile->addSeparator();
    m_menuFile->addAction( m_actionPreferences );
    m_menuFile->addSeparator();
    m_menuFile->addAction( m_actionExit );
    m_menuFile->setTitle( QApplication::translate( "MainWindow", "&File", 0,
            QApplication::UnicodeUTF8 ) );

    m_menuHelp->addAction( m_actionAbout_BrainCognize );
    m_menuHelp->setTitle( QApplication::translate( "MainWindow", "&Help", 0,
            QApplication::UnicodeUTF8 ) );

    m_glWidget = new BQtGLWidget();
    MainWindow->setCentralWidget( m_glWidget );

    // initially 3 views
    addDockableGLWidget( MainWindow );
    addDockableGLWidget( MainWindow );
    addDockableGLWidget( MainWindow );

    m_pipelineBrowser = new BQtPipelineBrowser();
    MainWindow->addDockWidget( Qt::RightDockWidgetArea, m_pipelineBrowser );

    MainWindow->setWindowTitle( QApplication::translate( "MainWindow",
            "BrainCognize", 0, QApplication::UnicodeUTF8 ) );
    m_toolBar->setWindowTitle( QApplication::translate( "MainWindow",
            "toolBar", 0, QApplication::UnicodeUTF8 ) );

    QObject::connect( m_actionExit, SIGNAL( activated() ), MainWindow,
            SLOT( close() ) );

    QMetaObject::connectSlotsByName( MainWindow );
}

void BMainWindow::setEnabled( bool enable )
{
    // TODO(wiebel) implement here
}
